js/battle.js

```diff
@@ -6104,19 +6104,24 @@ var Battle = (function () {
 				}
 				if (isSwitch && pokemon == this.p1.lastPokemon && !this.p1.active[slot]) continue;
 				if ((searchid && pokemon.searchid === searchid) || // exact match
-					(!pokemon.searchid && pokemon.checkDetails(details)) || // switch-in matches Team Preview entry
 					(!searchid && pokemon.ident === pokemonid)) { // name matched, good enough
 					if (!pokemon.searchid && createIfNotFound) {
```

Comment:
This entire block is unnecessary in this branch, actually.

Reply:
For the "exact match" case, `pokemon.searchid` will already exist, and for "name match only", not enough information is passed to actually create a Pokémon that isn't already found.